Makefile

```diff
@@ -342,13 +342,31 @@ shellcheck: $(SH_FILES) ## Run checks for shell scripts.
 TAG_SUFFIX=$(if $(WIRE_TAGS)!=oss,-$(WIRE_TAGS))
 PLATFORM=linux/amd64
 
+# default to a production build for frontend
+#
+DOCKER_JS_NODE_ENV_FLAG = production
+DOCKER_JS_YARN_BUILD_FLAG = build
+#
+# if go is in dev mode, also build node in dev mode
+ifeq ($(GO_BUILD_DEV), dev)
+  DOCKER_JS_NODE_ENV_FLAG = dev
```

Comment:
This will only build Grafana core - it won't build any of the decoupled datasource plugins. I think we need to update NXs `project.json` so it orchestrates builds of theme, icons and all plugins. This is possible by duplicating [the `build` target entry](https://github.com/grafana/grafana/blob/main/project.json#L16-L27) and renaming to dev.